pkg/counter/counter_test.go

```diff
@@ -66,95 +108,99 @@ func TestIncrease_MultipleThreads(t *testing.T) {
 
 	wg.Wait()
 
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncreaseWrapErr tests that the counter wraps when it reaches the maximum value.
-func TestIncreaseWrapErr(t *testing.T) {
+//
+// Decrement
+//
+
+func TestDecrement(t *testing.T) {
 	expected := uint64(0)
-	c := NewCounter(uint64(math.MaxUint64))
+	c := NewCounter(1)
 
-	err := c.Increase()
+	err := c.Decrement()
 
-	require.Error(t, err) // requires wrapped error
-	require.Equal(t, expected, c.Read())
+	require.NoError(t, err)
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncreaseMultiple tests that the counter is increased by the given values.
-func TestIncreaseMultiple(t *testing.T) {
-	expected := uint64(1 + 2 + 3)
-	c := NewCounter(0)
+func TestDecrementWithValue(t *testing.T) {
+	expected := uint64(1)
+	c := NewCounter(10)
 
-	err := c.Increase(1, 2, 3)
+	err := c.Decrement(9)
 
 	require.NoError(t, err)
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, c.Get())
 }
 
-// TestIncreaseMultipleSumWrapErr tests that the counter wraps when it reaches the maximum value.
-func TestIncreaseMultipleSumWrapErr(t *testing.T) {
-	expected := uint64(0)
-	c := NewCounter(expected)
+func TestDecrementWithMultipleValue(t *testing.T) {
+	expected := uint64(1)
+	c := NewCounter(10)
 
-	err := c.Increase(1, math.MaxUint64)
+	err := c.Decrement(3, 3, 3)
 
-	require.Error(t, err) // requires wrapped error
-	require.Equal(t, expected, c.Read())
+	require.NoError(t, err)
+	require.Equal(t, expected, c.Get())
 }
 
-//
-// Decrease
-//
-
-// TestZeroedDecrease tests that the counter is not decreased when 0 is given.
-func TestZeroedDecrease(t *testing.T) {
+func TestDecrementAndRead(t *testing.T) {
 	expected := uint64(1)
-	c := NewCounter(1)
+	c := NewCounter(10)
 
-	err := c.Decrease(0)
+	n, err := c.DecrementValueAndRead(9)
 
 	require.NoError(t, err)
-	require.Equal(t, expected, c.Read())
+	require.Equal(t, expected, n)
+	require.Equal(t, expected, c.Get())
 }
 
-// TestDecreaseAndRead tests that the counter is decreased by 1 and returns the new value.
-func TestDecreaseAndRead(t *testing.T) {
-	expected := uint64(0)
-	c := NewCounter(1)
+func TestZeroedDecrementWithValue(t *testing.T) {
+	expected := uint64(9)
+	c := NewCounter(9)
 
-	n, err := c.DecreaseAndRead()
+	err := c.Decrement(0)
```

Comment:
```suggestion
	err := c.Decrement(0, 0, 0
```

Reply:
![image](https://github.com/aquasecurity/tracee/assets/7395852/551ceb19-39e1-4a2f-82fa-91740b23d217)

I guess you want to test the "short circuit" for one or more zeroes, either way code works.

Reply:
Im adding this test and bencmark tests.

Reply:
Yes, I wanted to test the possible sum resulting in zero since the PR is decrementing val by 1 in the case of len(x) != 0, even sum being 0 (underflowing), what would result in a wrong inc/dec.

Reply:
I see the point, and despite the shortcircuit or not its worth testing that.